src/pip/_internal/configuration.py

```diff
@@ -344,6 +347,8 @@ def iter_config_files(self) -> Iterable[Tuple[Kind, List[str]]]:
             # The legacy config file is overridden by the new config file
             yield kinds.USER, config_files[kinds.USER]
 
+        yield kinds.BASE, config_files[kinds.BASE]
```

Comment:
> The "base" location should be skipped outside of virtual environments, where base_prefix and prefix are the same.

I considered this before opening the PR, but the problem is that we don't know what config files need to be loaded at this point - there may be a ``load-only`` of ``BASE``, which would mean we don't get that config file loaded in non-venv contexts. Perhaps that is OK though - I would add a caveat to the docs to say ``BASE`` is only considered when a virtual environment is active.

I was further dissuaded from my implementation by the todo at the top of the function, which asks for *less* logic in what config files to choose https://github.com/pypa/pip/pull/11487/files#diff-eadf79f17f8c7b56d3743fb4bb0ef3c2c2e062e493e8c25962f3319ef264f8bfR335.

> We should also provide a (temporary) opt-out from this, for users who might find this change to be disruptive.

I agree. I could add something specific to ``BASE``, or I could look into allowing the precedence order to be overridden (and to disallow certain levels). Something like:

```
config-precedence =
    env
    global
    user
    site
    env_var
```

That would amount to quite a big change though - and might require some thought about doing the config loading in two passes (first to determine the precedence, then to actually load it in the desired order) - this would reduce to a single pass in the case that the precedence *isn't* overridden. I would be happy to talk about this offline/elsewhere if this is something you would like me to pursue.

If you want me to keep it ``BASE`` scoped (and simpler), then suggestions are very welcome on the best approach - a simple env var, for example?

Reply:
I took a poke at implementing config-precedence config at https://github.com/pelson/pip/commit/0553eb1346c47b348eb33846be76ff77a1a92042. I think it turned out quite well in fact - might be a nice alternative option for allowing users to opt-out of ``BASE`` loading.

I would be happy to discuss this on a separate issue, or on discord, if this is interesting to pursue.

Reply:
Since the temporary opt-out comment is withdrawn in https://github.com/pypa/pip/pull/11487#issuecomment-1404104817, the remainging discussion is:

> The "base" location should be skipped outside of virtual environments, where base_prefix and prefix are the same.

In my response, I highlight that this particular function has a [comment](https://github.com/pypa/pip/blob/e69e265cb7b60fb2dacbbb2ab8fa3baaf24bfe4d/src/pip/_internal/configuration.py#L325):

```
# SMELL: Move the conditions out of this function
```

I agree with the suggestion that the function smells, as the function's objective should be to unconditionally enumerate the locations that configuration may come from, with ``_load_config_files`` determining if those files should be loaded or not.

For this reason, I added the BASE config file unconditionally. If we want to optimise and avoid BASE and SITE being loaded if they are the same, then I would put this logic in the ``_load_config_files`` method. Would be happy to add this, if it is considered important.